pkg/operator/sync.go

```diff
@@ -56,53 +57,80 @@ func (optr *Operator) syncAll(config *OperatorConfig) error {
 		return nil
 	}
 
+	errors := []error{}
 	// Sync webhook configuration
 	if err := optr.syncWebhookConfiguration(); err != nil {
+		errors = append(errors, fmt.Errorf("Error syncing machine API webhook configurations: %w", err))
+	}
+
+	if err := optr.syncClusterAPIController(config); err != nil {
+		errors = append(errors, fmt.Errorf("Error syncing machine-api-controller: %w", err))
+	}
+
+	// Sync Termination Handler DaemonSet if supported
+	if config.Controllers.TerminationHandler != clusterAPIControllerNoOp {
+		if err := optr.syncTerminationHandler(config); err != nil {
+			errors = append(errors, fmt.Errorf("Error syncing termination handler: %w", err))
+			return err
```

Comment:
did you meant to return early here?

Reply:
this doesn't make sense to me also, any clarification here Danil?